interview-questions.md

```diff
@@ -10,46 +10,81 @@ Then, research the question to expand on your answer. Even if you feel you have
+
 Researched answer:
 
+I would use most of my original answer. I would add that the term for all of these necessary packages and their versions are the "dependencies" for an application.
+
 3. What is a React component?
 
 Your answer:
 
+A component is React's building block for a complete application. React believes in the "separation of concerns" where parts of an app and their accociated files are kept as separate parts of a while as opposed to one large, difficult to manage file that holds the entire application in one place. A portion of a React application that performs a particular function or takes up a certain portion of a page is called a component. The components are then assembled into a functioning application. Components are linked to each other through properties and functions, so components often times cannot function independently without data or help from another componenet, but logically each component is its own entity that serves a singular purpose. 
+
 Researched answer:
 
+I would use most of my original answer and add that a major advantage of components comes with the ideology of "don't repeat yourself" (DRY). Once a component is built, it can be resued indefinitely with very little extra work. Componenet keep applications tidy and make future development much easier, as components can be added, removed, or updated with little disruption of the the rest of the application
+
 4. What is the difference between state values and props in React?
 
 Your answer:
 
+State and props are both major aspects of how React works. State is a form of memory within React. State allows parts of React components to have a default state and then change that state later depending on time, actions, or the state of another portion of the components or the state of something in another component. States can include the position of a switch, the value of a counter, or pretty much anything that is not static. The big reason this is important in React is that it allows changes without rendering the entire page for the change. If state changes withing a component, only that component changes. The rest of the page remains how it is. Props (short for properties) allows separate components of a React application to communicate with each other. While props only allows information to flow downhill (from higher components to lower components), it makes the component architecture possible. Without props, it would be impossible to pass data from one component to the other which would defeat the purpose of components in the first place. Props reinforces the idea of DRY by letting their be a single source of data which is then shared everywhere that data is shares through props.
```

Comment:
You are a React wizard. Do you enjoy it? You have a really good grasp on all of it's moving parts